runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -3710,21 +3710,20 @@ inline void generateInlinedCheckCastOrInstanceOfForInterface(TR::Node* node, TR_
 
    // Profiled call site cache
    uintptr_t guessClass = 0;
-   TR_OpaqueClassBlock* guessClassArray[NUM_PICS];
-   auto num_PICs = TR::TreeEvaluator::interpreterProfilingInstanceOfOrCheckCastInfo(cg, node, guessClassArray);
-   auto fej9 = (TR_J9VMBase *)(comp->fe());
-   for (uint8_t i = 0; i < num_PICs; i++)
+   if (!comp->compileRelocatableCode())
       {
-      if (fej9->instanceOfOrCheckCast((J9Class*)guessClassArray[i], (J9Class*)clazz))
+      TR_OpaqueClassBlock* guessClassArray[NUM_PICS];
+      auto num_PICs = TR::TreeEvaluator::interpreterProfilingInstanceOfOrCheckCastInfo(cg, node, guessClassArray);
+      auto fej9 = (TR_J9VMBase *)(comp->fe());
```

Comment:
Let's use C++ style casts where possible. They have the advantage of being compiler checked. Similarly below.

Reply:
Made changes to use C++ style cast https://github.com/eclipse/openj9/compare/32cebb26bab96e1e2dae7023f04be4ea5ac833ee..2299020f96bf446ca21f325cc3ff7c8e3f412281